cryptolib/cryptolib.go

```diff
@@ -44,30 +52,41 @@ func GetQuoteForCryptoAsset(priceFeeds []*PriceFeed, asset string) (*messagelib.
 	return &messagelib.TickerValue{Ticker: asset, Value: float32(price), Change: float32(change) * 100.0}, nil
 }
 
-func getPriceFeed(priceFeeds []*PriceFeed, asset string) (*PriceFeed, bool) {
-	for i := range priceFeeds {
-		if priceFeeds[i].Pair == asset {
-			return priceFeeds[i], true
+func getFeedForAsset(geminiClient *http.Client, asset string) (*PriceFeed, bool) {
+	fetchPriceFeeds(geminiClient)
+	for _, feed := range priceFeeds {
+		if feed.Pair == asset {
+			return feed, true
 		}
 	}
 	return nil, false
 }
 
-// GetPriceFeeds returns the current Price points of cryptos traded on Gemini.
-func GetPriceFeeds(geminiClient *http.Client) ([]*PriceFeed, error) {
-	var priceFeeds []*PriceFeed
+func fetchPriceFeeds(geminiClient *http.Client) {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if time.Since(lastUpdated) <= priceFeedAgeLimit {
+		return
+	}
+
+	log.Printf("Crypto price feeds are older than %v, fetching update.", priceFeedAgeLimit)
+
+	var newPriceFeeds []*PriceFeed
```

Comment:
Maybe a future improvement (file an issue if you agree):
Should we keep track of a bit that says "Attempted to fetch the most recent price feeds, but failed, so we will use stale data"?

We could then use that bit to modify the message embed with something along the lines of "Data might be stale" so folks know to go check their own coinbase profile or gemini profile for most recent exchange prices